pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -548,13 +548,14 @@ func (exp *flowExporter) addRecordToSet(record flowexporter.FlowRecord) error {
 		case "flowType":
 			ie.Value = exp.findFlowType(record.Conn)
 		}
+		eL[i] = ie
```

Comment:
I don't know if that makes a copy, but I image so and to be safe you should use:
```go
for i := range eL {
      ie := &eL[i]
      // ...
}
```

same comment for other loop below

Reply:
Any reference for this? Thanks.

Reply:
it's not about correctness, but you are copying the element back into the list so it's slower for no reason

see this benchmark:
```go
package main

import (
	"testing"
)

// Without Big
// BenchmarkCopy-12    	 1000000	      1151 ns/op
// BenchmarkRef-12     	 1000000	      1027 ns/op

// With Big
// BenchmarkCopy-12    	   46578	     23557 ns/op
// BenchmarkRef-12     	  869930	      1425 ns/op

type Data struct {
	A   int64
	B   int64
	C   int64
	D   int64
	Big [32]int64
}

func makeList(N int) []Data {
	return make([]Data, N)
}

func iterCopy(l []Data) []Data {
	for i, e := range l {
		e.A += 1
		e.B += 1
		e.C += 1
		e.D += 1
		l[i] = e
	}
	return l
}

func iterRef(l []Data) []Data {
	for i := range l {
		e := &l[i]
		e.A += 1
		e.B += 1
		e.C += 1
		e.D += 1
	}
	return l
}

func BenchmarkCopy(b *testing.B) {
	l := makeList(1000)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		iterCopy(l)
	}
}

func BenchmarkRef(b *testing.B) {
	l := makeList(1000)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		iterRef(l)
	}
}
```

The performance difference can vary significantly based on element size.

Reply:
I missed the performance reference in your comment.
Here the object is not big. It has a pointer and value of type interface. As you mentioned, better to be safe.
Thanks for the catch.